### DISCUSSION TITLE
What APIs should be included in the standard library?

### DISCUSSION THREAD

**Discussion by Chriscbr** [winglang/wing#6487] [General]
What are the APIs in Wing's standard library?

It's possible to access most Node APIs and third-party library functions through [extern functions](https://www.winglang.io/docs/language-reference#511-external-libraries). But there should also be a selection of APIs that come pre-included with Wing.

Today, we already have several built-in modules like `http`, `math`, and `util`. What set of principles should we use for deciding which APIs to include vs exclude? Offering APIs in Wing has the upside of providing a better experience for first time users (since they don't need to write JavaScript or install third party libraries), but it also increases our API surface area that needs to be maintained.

Common standard library features in other programming languages for inspiration:
<details>
- IO (input/output), stdout/stderr/stdin, argument parsing
- random (generating numbers, shuffling, etc.)
- hashing / cryptographic functions
- debugging functions
- error types?
- regular expressions
- file system
- networking (fetch? sockets? IP utilities?)
- OS (number of CPUs, platform, creating temporary files/directories)
- path manipulation
- URL manipulation
- timers (sleep, set timeout, time elapsed)
- reflection
- meta (current filename or directory name?)
- processing (forks, PIDs)
- threading
- string formatting
- stack traces
- type converting utilities, type narrowing utilities (isInt...)
- iteration utilities (permutations, zipping, cycles, ...)
- FFI
- math
- logging
- serialization
</details>

[a comment by an automated account is not shown]

**Comment by Chriscbr** [comment]
Initial set of APIs added in this RFC: https://github.com/winglang/wing/pull/2837/files

[a comment by an automated account is not shown]

**Comment by staycoolcall911** [comment]
Chatted with @Chriscbr and we decided to convert this issue into a discussion, since it is not really actionable
